klog.go

```diff
@@ -874,8 +882,24 @@ func (sb *syncBuffer) Sync() error {
 	return sb.file.Sync()
 }
 
+var realMaxSizeCalc = func () uint64 {
+	if logging.logFile != "" {
+		if logging.logFileMaxSize == 0 {
+			// If logFileMaxSize is zero, we don't have limitations on the log size.
+			return math.MaxUint64
+		} else {
+			// Flag logFileMaxSize is in MB for user convenience.
+			return logging.logFileMaxSize * 1024 * 1024
+		}
+	}
+	// If "log_file" flag is not specified, the target file (sb.file) will be cleaned up when reaches a fixed size.
+	return MaxSize
+}
+
+var PrecomputedMaxSize = realMaxSizeCalc()
```

Comment:
This needs to happen after flag.Parse(), so it cannot be done at init time. I think it might make sense to make it a field on  syncBuffer.